src/test/groovy/com/stormpath/tck/register/RegisterIT.groovy

```diff
@@ -0,0 +1,576 @@
+        def results = new ArrayList<Node>()
+
+        for (Node node in children.list()) {
+            if (node.name() == tag) {
+                results.add(node)
+            }
+            else {
+                Collection<Node> innerResults = findTags(node.children(), tag)
+                results.addAll(innerResults)
+            }
+        }
+
+        return results
+    }
+
+    /**
+     * Serve the registration view model for request type application/json
+     * @see <a href="https://github.com/stormpath/stormpath-framework-tck/issues/179">#179</a>
+     * @throws Exception
+     */
+    @Test
+    public void servesViewModel() throws Exception {
+
+        given()
+            .accept(ContentType.JSON)
+        .when()
+            .get(registerRoute)
+        .then()
+            .statusCode(200)
+            .contentType(ContentType.JSON)
+            .body("size()", is(2))
+            .body(".", hasKey("form"))
+            .body(".", hasKey("accountStores"))
+    }
+
+    /**
+     * Return JSON error if required fields are missing
+     * @see <a href="https://github.com/stormpath/stormpath-framework-tck/issues/189">#189</a>
+     * @throws Exception
+     */
+    @Test
+    public void returnsErrorIfPostIsEmpty() throws Exception {
+
+        given()
+            .accept(ContentType.JSON)
+        .when()
+            .post(registerRoute)
+        .then()
+            .statusCode(400)
+            .contentType(ContentType.JSON)
+            .body("status", is(400))
+            .body("message", not(isEmptyOrNullString()))
+    }
+
+    /**
+     * Return JSON error if required fields are missing
+     * @see <a href="https://github.com/stormpath/stormpath-framework-tck/issues/189">#189</a>
+     * @throws Exception
+     */
+    @Test
+    public void returnsErrorIfEmailIsMissing() throws Exception {
+
+        Map<String, Object>  jsonAsMap = new HashMap<>();
+        jsonAsMap.put("email", accountEmail)
+        jsonAsMap.put("password", "")
+
+
+        given()
+            .accept(ContentType.JSON)
+            .body(jsonAsMap)
+        .when()
+            .post(registerRoute)
+        .then()
+            .statusCode(400)
+            .contentType(ContentType.JSON)
+            .body("status", is(400))
+            .body("message", not(isEmptyOrNullString()))
+    }
+
+    /**
+     * Return JSON error if required fields are missing
+     * @see <a href="https://github.com/stormpath/stormpath-framework-tck/issues/189">#189</a>
+     * @throws Exception
+     */
+    @Test
+    public void returnsErrorIfRequiredFieldIsMissing() throws Exception {
+
+        Map<String, Object>  jsonAsMap = new HashMap<>();
+        jsonAsMap.put("email", accountEmail)
+        jsonAsMap.put("password", accountPassword)
+        jsonAsMap.put("surname", accountSurname)
+        // givenName is required per the default configuration
+
+        given()
+            .accept(ContentType.JSON)
+            .body(jsonAsMap)
+        .when()
```

Comment:
I logged this, and it's not doing things correctly:

```
Request method:	POST
Request path:	http://localhost:3000/register
Proxy:			<none>
Request params:	<none>
Query params:	<none>
Form params:	<none>
Path params:	<none>
Multiparts:		<none>
Headers:		Accept=application/json, application/javascript, text/javascript
				Content-Type=text/plain; charset=ISO-8859-1
Cookies:		<none>
Body:
{"password":"P@sword123!","surname":"Surname-c8cc12da-4928-454e-ad66-233479bb9d87","givenName":"GivenName-c8cc12da-4928-454e-ad66-233479bb9d87","middleName":"Foobar","email":{"values":["fooemail-c8cc12da-4928-454e-ad66-233479bb9d87@stormpath.com"],"strings":["json-",""],"bytes":"anNvbi1mb29lbWFpbC1jOGNjMTJkYS00OTI4LTQ1NGUtYWQ2Ni0yMzM0NzliYjlkODdAc3Rvcm1wYXRoLmNvbQ==","valueCount":1},"username":"foo-c8cc12da-4928-454e-ad66-233479bb9d87"}
```

Let's set the content type properly.